### PR TITLE
OPSEXP-2296 Review and align chart alfresco-connector-msteams with newer principals

### PR DESCRIPTION
**Pull request by slohe1**

### CODE REVIEW

File: charts/alfresco-connector-msteams/templates/deployment-connector-msteams.yaml [hunk 1]
```diff
@@ -21,11 +21,11 @@ spec:
       labels:
         {{- include "alfresco-connector-msteams.selectorLabels" . | nindent 8 }}
     spec:
-    {{- include "component-pod-security-context" .Values | indent 4 }}
+    {{- include "alfresco-common.component-pod-security-context" .Values | indent 4 }}
```
Comment (alxgomz) [inline charts/alfresco-connector-msteams/templates/deployment-connector-msteams.yaml:24]: please add missing unit tests

File: charts/alfresco-connector-msteams/templates/deployment-connector-msteams.yaml [hunk 2]
```diff
@@ -21,11 +21,11 @@ spec:
...
     {{- if .Values.nodeSelector }}
       nodeSelector: {{- .Values.nodeSelector | toYaml | nindent 8 }}
     {{- end }}
-      {{- include "alfresco-content-services.imagePullSecrets" . | indent 6 }}
+      {{- include "alfresco-common.imagePullSecrets" . | indent 6 }}
```
Comment (alxgomz) [inline charts/alfresco-connector-msteams/templates/deployment-connector-msteams.yaml:28]: please add missing unit tests

File: charts/alfresco-connector-msteams/templates/deployment-connector-msteams.yaml [hunk 3]
```diff
@@ -59,7 +59,7 @@ spec:
         - name: {{ .Chart.Name }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
-          {{- include "component-security-context" .Values.msTeams | indent 8 }}
+          {{- include "alfresco-common.component-security-context" .Values | indent 8 }}
```
Comment (alxgomz) [inline charts/alfresco-connector-msteams/templates/deployment-connector-msteams.yaml:62]: please add missing unit tests

**Review by alxgomz — CHANGES_REQUESTED** [review]
The following things are mising (please see the JIRA):
* chart.serviceAccount template
* chart offers the ability to provide configuration of other components of the platform through configmap

File: charts/alfresco-connector-msteams/templates/secret-msteams.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+apiVersion: v1
+kind: Secret
+metadata:
+  {{- $secretCtx := dict "Values" (dict "nameOverride" "msteams-se") "Chart" .Chart "Release" .Release }}
+  name: {{ template "alfresco-connector-msteams.fullname" $secretCtx }}
+  labels:
+    {{- include "alfresco-connector-msteams.labels" . | nindent 4 }}
+type: Opaque
+data:
+  MICROSOFT_APP_ID: {{ .Values.microsoft.app.id  }}
```
Comment (gionn) [inline charts/alfresco-connector-msteams/templates/secret-msteams.yaml:11]:
```suggestion
  MICROSOFT_APP_ID: {{ .Values.microsoft.app.id }}
```

**Review by gionn — COMMENTED** [review]
to fix the tests add some random value to the required values in ci/default-values.yaml

File: charts/alfresco-connector-msteams/Chart.yaml [hunk 5]
```diff
@@ -2,9 +2,9 @@ apiVersion: v2
 name: alfresco-connector-msteams
 description: A Helm chart for deploying Alfresco connector msteams service
 type: application
-version: 0.2.0
+version: 0.2.1-alpha.0
```
Comment (gionn) [inline charts/alfresco-connector-msteams/Chart.yaml:5]: you are adding new features which are retrocompat, so it's a minor version bump, not a patch

File: charts/alfresco-connector-msteams/templates/secret-msteams.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+{{- if not .Values.microsoft.app.existingSecret.name }}
+apiVersion: v1
+kind: Secret
+metadata:
+  {{- $secretCtx := dict "Values" (dict "nameOverride" "msteams-se") "Chart" .Chart "Release" .Release }}
+  name: {{ template "alfresco-connector-msteams.fullname" $secretCtx }}
```
Comment (gionn) [inline charts/alfresco-connector-msteams/templates/secret-msteams.yaml:6]: it's better to have a named helper so you avoid this copy-pasta in multiple places, like I did [here](https://github.com/Alfresco/alfresco-helm-charts/pull/175/files#diff-cead3e4fd48780ae4d8f8e2cc4fb668b1e73bc40a836a1a143c13170e36b98fa)

File: charts/alfresco-connector-msteams/tests/configmap_test.yaml [hunk 7]
```diff
@@ -0,0 +1,33 @@
+suite: test msteams-connector configmap
+templates:
+- config-connector-msteams.yaml
+- configmap-repository.yaml
```
Comment (gionn) [inline charts/alfresco-connector-msteams/tests/configmap_test.yaml:4]: I prefer we keep unrelated resources in separated test suite files, e.g. config-connector-msteams_test.yaml and configmap-repository_test.yaml

in that way you can also avoid having to repeat `template` in every test

File: charts/alfresco-connector-msteams/Chart.yaml [hunk 8]
```diff
@@ -2,9 +2,9 @@ apiVersion: v2
 name: alfresco-connector-msteams
 description: A Helm chart for deploying Alfresco connector msteams service
 type: application
-version: 0.2.0
+version: 0.2.1
```
Comment (gionn) [inline charts/alfresco-connector-msteams/Chart.yaml:5]:
```suggestion
version: 0.3.0-alpha.0
```

**Review by gionn — CHANGES_REQUESTED** [review]

**Review by gionn — APPROVED** [review]

**Pull request merged**